R/gargle-package.R

```diff
@@ -92,3 +92,25 @@ gargle_oob_default <- function() {
 gargle_oauth_cache <- function() {
   getOption("gargle_oauth_cache", default = NA)
 }
+
+#' @rdname gargle_options
+#' @export
+#' @section `gargle_oauth_client_type`:
+#' `gargle_oauth_client_type()` returns the option named
+#' "gargle_oauth_client_type", if defined. If defined, the option must be either
+#' "installed" or "web". If the option is not defined, the function returns:
+#' * "web" on RStudio Server, Posit Workbench, Posit Cloud, or Google
+#'   Colaboratory
+#' * "installed" otherwise
```

Comment:
Why is this rigged differently from `gargle_oob_default()`, where we impose our own logic *before* consulting an option? Here we first consult the option and impose our own logic if the option is undefined.

Because it seems like we must do some form of OOB auth on RStudio Server, Posit Workbench, Posit Cloud, or Google Colab. Starting a local webserver will never make sense in this context.

Whereas, once we know we're doing some form of OOB, the client type determines whether it's conventional OOB ("installed" client type) or pseudo-OOB ("web" client type). I expect conventional OOB to be relatively rare these days, but it is still possible.